inc/admin/namespace.php

```diff
@@ -245,6 +240,47 @@ function handle_edit_submit( Client $consumer = null ) {
 	exit;
 }
 
+function enqueue_javascript() {
+	wp_localize_script('oauth2-edit-application', 'oauth2_ajax', [
+		'action' => AJAX_GENERATE_ACCESS_TOKEN,
+		'url' => admin_url( 'admin-ajax.php' ),
+	]);
+	wp_enqueue_script( 'oauth2-edit-application' );
+}
+
+function get_application_id() {
+	if ( ! empty( $_SERVER['HTTP_REFERER'] ) ) {
+		$id = \parse_url( $_SERVER['HTTP_REFERER'], PHP_URL_QUERY );
+		$params = [];
+		parse_str( $id, $params );
+		if ( ! isset( $params['id'] ) ) {
+			throw 'Not on admin page of the Application';
```

Comment:
This needs to be `throw new Exception`, but I'd rather use `return new WP_Error` instead.